File: src/main/java/it/alten/pokemonao/models/PokemonModel.java
package it.alten.pokemonao.models;

import lombok.Data;

@Data
public class PokemonModel {
    private Integer id;
    private String name;
    private String sprite;
    private int currentHp;
    private int maxHp;
    private TypeModel type;
    private MoveModel[] moves;
    private String trainerName;
}

File: pokemon-ao-frontend/src/app/shared/models/pokemon.model.ts
import {TypeModel} from "./type.model";
import {MoveModel} from "./move.model";

export interface PokemonModel {
  pokemonPokeApiId: number;
  nickname: string;
  sprite: string;
  currentHp: number;
  maxHp: number;
  type: TypeModel;
  moves: MoveModel[];
}
